### PR TITLE
Add code for supporting GooglePay and ApplePay tokens. Made methods p…

### PR DESCRIPTION
**Pull request by ashakoorspecialized**
…rotected in AdyenRequestFactory so that they can be extended in custom code

**Description**

**Tested scenarios**

**Fixed issue**:

### CODE REVIEW

**Comment by rkewlani** [comment]
Thanks @ashakoorspecialized for PR.
We will review it and let you know our feedbacks.

Regards,
Rajni 
Adyen

**Comment by rkewlani** [comment]
Hi @ashakoorspecialized , I can see this is only backend code. Did you also make front end changes ?
Thanks,
Rajni

**Comment by ashakoorspecialized** [comment]
No front end code changes were made. We are directly using Googlepay/Applepay Js libraries for front end.

File: adyenv6core/src/com/adyen/v6/factory/AdyenRequestFactory.java [hunk 1]
```diff
@@ -766,16 +777,44 @@ private void setBoletoData(PaymentsRequest paymentsRequest, CartData cartData) {
...
+     * Set Applepay request data
+     */
+    protected void setApplePayData(PaymentsRequest paymentsRequest, CartData cartData) {
+        DefaultPaymentMethodDetails paymentMethodDetails = (DefaultPaymentMethodDetails) (paymentsRequest.getPaymentMethod());
+        if (paymentMethodDetails == null) {
+            paymentMethodDetails = new DefaultPaymentMethodDetails();
+        }
+
+        paymentMethodDetails.setType(APPLEPAY);
+        paymentMethodDetails.setGooglepayToken(cartData.getAdyenApplepayToken());
```
Comment (Aleffio) [inline adyenv6core/src/com/adyen/v6/factory/AdyenRequestFactory.java]:
```suggestion
        paymentMethodDetails.setApplepayToken(cartData.getAdyenApplepayToken());
```

Comment (ashakoorspecialized) [reply adyenv6core/src/com/adyen/v6/factory/AdyenRequestFactory.java]: Great catch :)

**Comment by AlexandrosMor** [comment]
Hello @ashakoorspecialized,

There are some conflicts to resolve, It would be nice if you can fix them before we merge your changes.

Thank you in advance,
Alexandros

**Comment by rkewlani** [comment]
Hi @ ashakoorspecialized,
Sorry for late response. We were in internal discussions about this change and we reached to a conclusion that we will close this PR for now because it is incomplete from the plugin perspective as it includes only backend logic. We are planning to support google pay soon so we will reuse your code at that time. Also applepay will have a new approach in upcoming year so for now we are not in state to merge it for applepay either. thanks for your time and efforts in this.
Reagrds,
Adyen

**Comment by ashakoorspecialized** [comment]
That is disappointing. We will continue to use our forked branch for the foreseeable future to support our needs.

**Pull request closed without merging**